fritzconnection/core/fritzconnection.py

```diff
@@ -294,3 +294,14 @@ def reboot(self):
         Reboot the system.
         """
         self.call_action("DeviceConfig1", "Reboot")
+
+    def device_support_mesh(self):
+        """
+        Return True if the device support mesh.
+        """
+        if (
+            "Hosts1" not in self.services
+            or "X_AVM-DE_GetMeshListPath" not in self.services["Hosts1"].actions
+        ):
+            return False
+        return True
```

Comment:
This should also do it 🤔 
```suggestion
        return not (
            "Hosts1" not in self.services
            or "X_AVM-DE_GetMeshListPath" not in self.services["Hosts1"].actions
        )
```

Reply:
We can get rid of all `not` and move "`or`" to "`and`"

Reply:
I think this is a useful information about a router capability to make it into the core. However I would implement it a little bit different and also this should get testet - even if it is necessary to mock the router.

```python
MESH_SERVICE_NAME = "Hosts1"
MESH_ACTION_NAME = "X_AVM-DE_GetMeshListPath"

...

@property
def has_mesh_support(self):
    try:
        return MESH_ACTION_NAME in self.services[MESH_SERVICE_NAME].actions
    except KeyError:
        return False
```